monitoring/router.nginx.conf

```diff
@@ -1,17 +1,17 @@
 server {
     server_name _;
 
-    location /monitoring/grafana/ {
+    location ~ ^/monitoring/grafana(/)?$ {
```

Comment:
You can just remove the trailing slash.

Reply:
but then it would match http://domain2.com/monitoring/grafanaddd

Reply:
Specifically without a = nginx will perform a prefix match

Reply:
Specifically, location blocks without a modifier match the URL prefix.

Reply:
I think I don't understand how this is different from what I wrote.

Reply:
This won't allow a path after the prefix, e.g. /monitoring/grafana/foo/bar.  I think what you'd want it `^/monitoring/grafana($|/)`.  However, the point of setting Host in the gateway to include the target service is exactly so we can use it to dispatch in nginx.  So I'd change this to use separate server blocks like we do in other routers.